oof/src/lib.rs

```diff
@@ -220,6 +222,43 @@ mod tests {
         args.map(OsString::from).collect()
     }
 
+    fn setup_args_scenario(arg_str: &str) -> Result<(Vec<OsString>, Flags), OofError> {
+        let flags_info = [
+            ArgFlag::long("output_file").short('o'),
+            Flag::long("verbose").short('v'),
+            Flag::long("help").short('h'),
+        ];
+
+        let args = gen_args(arg_str);
+        filter_flags(args, &flags_info)
+    }
+
+    #[test]
+    fn test_unknown_flags() {
+        let result = setup_args_scenario("ouch a.zip -s b.tar.gz c.tar").unwrap_err();
+        assert!(matches!(result, OofError::UnknownShortFlag('s')));
+        let result = setup_args_scenario("ouch a.zip --foobar b.tar.gz c.tar").unwrap_err();
+        // TODO: Try removing the `_` from `_foobar`
```

Comment:
I don't quite understand what is happening here with this `matches!` macro. Please attempt to fullfill these TODOs and see what happens :smile:

Reply:
`matches!` does not compare the fields, it only tries to pattern match against the enum variants, the argument you're passing, for example: `_duplicated_flag`, does not change the pattern.

Look at this this example:
```rust
let option: Option<i32> = Some(10);
match option {
    Some(value) => println!("{}", value),
    None => {},
}
```

`value` will be a binding to the value that is matched, now look at this other example:
```rust
let option: Option<i32> = Some(10);
let value = 102;
match option {
    Some(value) => println!("{}", value),
    None => {},
}
```

This code is exactly the same as the previous, `value` is not being used, the pattern is the same.